kong/init.lua

```diff
@@ -1427,6 +1443,7 @@ function Kong.log()
   execute_collected_plugins_iterator(plugins_iterator, "log", ctx)
   runloop.log.after(ctx)
 
+  release_table(CTX_NS, ctx)
```

Comment:
@kikito afaik we don't `pcall` the `log` plugins, so it may be possible that we don't release, if the log plugin runtime crashes. Am I worried by it? Not really. At some point we should do proper error handling, which has been a long in back logs.

Reply:
Ok I gave this another look and I think I understood.

`fetch` never increases the pool size, only decreases it - or simply creates a new table and returns it. `release_table` is actually the one which "increases the size of the pool" (once it gets called, the no-longer-used table goes to the pool). So if we get into a scenario where a plugin is always throwing an error in the log phase then `release_table` will never be called, but that won't mean we have a memory leak. It will mean that we have the same thing that we have now - a new table allocation on every request. ok then.